src/hybridmap/HybridNeutralAtomMapper.cpp

```diff
@@ -37,136 +43,200 @@
 #include <vector>
 
 namespace na {
-qc::QuantumComputation NeutralAtomMapper::map(qc::QuantumComputation& qc,
-                                              InitialMapping initialMapping) {
-  mappedQc = qc::QuantumComputation(arch.getNpositions());
-  nMoves = 0;
-  nSwaps = 0;
+void NeutralAtomMapper::mapAppend(qc::QuantumComputation& qc,
+                                  const Mapping& initialMapping) {
+  // check if multi-qubit gates are present
+  multiQubitGates = false;
+  for (const auto& op : qc) {
+    if (op->getUsedQubits().size() > 2) {
+      // deactivate static mapping
+      multiQubitGates = true;
+      break;
+    }
+  }
+  // only add flying ancillas if not already present
+  if (mappedQc.getNancillae() == 0) {
+    // ancilla register has indices [npositions, 2*npositions-1]
+    mappedQc.addAncillaryRegister(this->arch->getNpositions());
+    // flying ancilla register has indices [2*npositions, 3*npositions-1]
+    mappedQc.addAncillaryRegister(this->arch->getNpositions(), "fa");
+  }
+
   qc::CircuitOptimizer::replaceMCXWithMCZ(qc);
   qc::CircuitOptimizer::singleQubitGateFusion(qc);
   qc::CircuitOptimizer::flattenOperations(qc);
   qc::CircuitOptimizer::removeFinalMeasurements(qc);
 
-  auto dag = qc::CircuitOptimizer::constructDAG(qc);
+  const auto dag = qc::CircuitOptimizer::constructDAG(qc);
 
-  // init mapping
-  this->mapping = Mapping(qc.getNqubits(), initialMapping);
+  mapping = initialMapping;
+
+  if (this->parameters.verbose) {
+    std::cout << "* Init Coord Mapping w/ [row:" << arch->getNrows()
+              << " X col:" << arch->getNcolumns() << "] hardware" << "\n";
+    for (uint32_t q = 0; q < qc.getNqubits(); q++) {
+      std::cout << "q " << std::setw(3) << q;
+      const auto hwQubit = this->mapping.getHwQubit(q);
+      std::cout << " -> h " << std::setw(3) << hwQubit;
+      std::cout << " -> c " << std::setw(3)
+                << hardwareQubits.getCoordIndex(hwQubit) << "\n";
+    }
+    std::cout << "\n";
+  }
```

Comment:
Holds for the entire file here: could it be worth to consistenly use SPDLOG here instead of standard output?